### PR TITLE
Created Notification class to set notification payload and Added Cache Manager

### PR DESCRIPTION
**Pull request by aishwa8141**

### CODE REVIEW

File: ui-sample/util-service/EPRFunctions.js [hunk 1]
```diff
@@ -1,36 +1,72 @@
...
 class EPRFunctions extends Functions {
     EPRFunctions() {
         setRequest(undefined)
-        
     }
 
     getAdminUsers(callback) {
-        this.getUsersByRole('admin', callback)
+        this.getUsersByRole('admin', (err, data) => {
+            this.addToPlaceholders('emailIds', _.map(data, 'email'))
```
Comment (indrajra) [inline ui-sample/util-service/EPRFunctions.js]: can you carve out a single function for these two lines that seem to be repeated?
I want the key 'emailIds' to be repeated less often mainly and the 'email'.

File: ui-sample/util-service/EPRFunctions.js [hunk 2]
```diff
@@ -40,14 +76,40 @@ class EPRFunctions extends Functions {
...
+    invoke(actions, callback2) {
+        if (actions.length > 0) {
+            let count = 0;
+            async.forEachSeries(actions, (value, callback) => {
+                count++;
+                this[value]((err, data) => {
+                    callback()
+                });
+                if (count == actions.length) {
+                    callback2(null, actions);
```
Comment (indrajra) [inline ui-sample/util-service/EPRFunctions.js]: usual practice is to use suffix 2 in the internal functions. To be more explicit, swap 'callback' and 'callback2' names.

File: ui-sample/util-service/app.js [hunk 3]
```diff
@@ -15,14 +15,20 @@ const keycloakHelper = require('./sdk/keycloakHelper');
 const logger = require('./sdk/log4j');
 const port = process.env.PORT || 9081;
 let wfEngine = undefined
+var CacheManager = require('./sdk/cacheManager.js');
+var cache_config = {
+    store: 'memory',
+    ttl: 1800
```
Comment (indrajra) [inline ui-sample/util-service/app.js]: put a comment before this var - what is ttl unit - seconds or milliseconds?

File: ui-sample/util-service/sdk/notification.js [hunk 4]
```diff
@@ -1,53 +1,62 @@
...
-    const option = {
-        method: 'POST',
-        url: notificationHost,
-        headers: {
-            'content-type': 'application/json',
-            'accept': 'application/json'
-        },
-        body: reqBody,
+class Notification {
+    constructor(mode, deliveryType, placeholder) {
```
Comment (indrajra) [inline ui-sample/util-service/sdk/notification.js]: if you type /** you will get a comment block, where you can explain what values can be passed. For example, mode can be email, sms.

File: ui-sample/util-service/sdk/notification.js [hunk 5]
```diff
@@ -1,53 +1,62 @@
...
-    httpUtil.post(option, function (err, res) {
-        if (res) {
-            logger.info("notification has been sucessfully sent", res.body)
-            callback(null, res.body)
-        } else {
-            logger.error("sending notification is unsuccessfull", err)
-            callback(err)
+
+    sendNotifications(callback) {
+        logger.info("email ids to send notifications", this.placeholder);
```
Comment (indrajra) [inline ui-sample/util-service/sdk/notification.js]: remove printing placeholder - emails are sensitive and we must not leak it even in logs.

File: ui-sample/util-service/workflow/Functions.js [hunk 6]
```diff
@@ -2,10 +2,15 @@ const _ = require('lodash')
...
+const Notification = require('../sdk/notification.js')
 const registryService = require('../sdk/registryService.js');
-
 const logger = require('../sdk/log4j.js')
+var CacheManager = require('../sdk/cacheManager.js');
+var cache_config = {
+    store: 'memory',
+    ttl: 1800
+}
+var cacheManager = new CacheManager(cache_config);
```
Comment (indrajra) [inline ui-sample/util-service/workflow/Functions.js]: can the CacheManager take nothing and have a setConfig method for people that wish to change. You seem to be needing cache_config in at least 2 files now.

File: ui-sample/util-service/sdk/notification.js (hunk 4 above)
Comment (aishwa8141) [reply ui-sample/util-service/sdk/notification.js]: done

File: ui-sample/util-service/EPRFunctions.js (hunk 1 above)
Comment (aishwa8141) [reply ui-sample/util-service/EPRFunctions.js]: I had created addEmailToPlaceHolderfunction to remove repetition of emailIds and email

File: ui-sample/util-service/EPRFunctions.js (hunk 2 above)
Comment (aishwa8141) [reply ui-sample/util-service/EPRFunctions.js]: done

File: ui-sample/util-service/sdk/notification.js (hunk 5 above)
Comment (aishwa8141) [reply ui-sample/util-service/sdk/notification.js]: removed

File: ui-sample/util-service/workflow/Functions.js (hunk 6 above)
Comment (aishwa8141) [reply ui-sample/util-service/workflow/Functions.js]: I had written setConfig method to set cache config

**Pull request merged**